tests/e2e/multi_vc_connection.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	neturl "net/url"
+	"strings"
+
+	gomega "github.com/onsi/gomega"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/vim25"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+connectMultiVC helps make a connection to a multiple vCenter Server. No actions are taken if a connection
+exists and alive. Otherwise, a new client will be created.
+*/
+func connectMultiVC(ctx context.Context, vs *multiVCvSphere) {
+	clientLock.Lock()
+	defer clientLock.Unlock()
+	if vs.multiVcClient == nil {
+		framework.Logf("Creating new VC session")
+		vs.multiVcClient = newClientForMultiVC(ctx, vs)
+	}
+	for i := 0; i < len(vs.multiVcClient); i++ {
+		manager := session.NewManager(vs.multiVcClient[i].Client)
```

Comment:
I see that we are adding a for loop and object as making an array of objects but functionality remains same. can't we do in exiting utils itself?

Reply:
I don't think we can use existing code, the existing code is expecting struct  like this 

type vSphere struct {
	Config    *e2eTestConfig
	Client    *govmomi.Client
	CnsClient *cnsClient
}

MultiVC struct code is like below, list of VCs and list of CNSClients

type multiVCvSphere struct {
	multivcConfig    *e2eTestConfig
	multiVcClient    []*govmomi.Client
	multiVcCnsClient []*cnsClient
}

If I change existing utils, the entire e2e codebase will break and it will require rigorous testing... whole regression suite i need to change then